frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-popover.tsx

```diff
@@ -0,0 +1,123 @@
+import * as React from 'react';
+import { getName, getNamespace, ProgressStatus } from '@console/shared';
+import { InProgressIcon, ErrorCircleOIcon, BanIcon } from '@patternfly/react-icons';
+import {
+  Popover,
+  PopoverPosition,
+  Button,
+  Progress,
+  ProgressVariant,
+  Stack,
+  StackItem,
+  Spinner,
+} from '@patternfly/react-core';
+import { UPLOAD_STATUS } from './consts';
+import { CDIUploadContext } from './cdi-upload-provider';
+import { killUploadPVC } from '../../k8s/requests/cdi-upload/cdi-upload-requests';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+export const getProgressVariant = (status: string) => {
+  switch (status) {
+    case UPLOAD_STATUS.ERROR:
+      return ProgressVariant.danger;
+    case UPLOAD_STATUS.SUCCESS:
+      return ProgressVariant.success;
+    default:
+      return null;
+  }
+};
+
+export const UploadPVCPopover: React.FC<PVCUploadStatusProps> = ({ pvc }) => {
+  const pvcName = getName(pvc);
+  const namespace = getNamespace(pvc);
+  const { uploads } = React.useContext(CDIUploadContext);
+  const upload = uploads.find((upl) => upl.pvcName === pvcName && upl.namespace === namespace);
+  const [error, setError] = React.useState(upload?.uploadError);
+
+  React.useEffect(() => {
+    setError(upload?.uploadError);
+  }, [upload]);
+
+  const getPopoverBody = (status: string) => {
+    switch (status) {
+      case UPLOAD_STATUS.ERROR:
+        return {
+          title: `Upload Error`,
+          body: error?.message,
+          icon: <ErrorCircleOIcon className="co-icon-and-text__icon" color="#cf1010" />,
```

Comment:
would be good to use PF tokens, look at https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/status/icons.tsx#L14